tests/k8st/tests/test_bgp_advert.py

```diff
@@ -430,20 +467,64 @@ def test_mainline(self):
             except subprocess.CalledProcessError:
               pass
 
+
+            # Get host IPs for the nginx pods.
+            localSvcHostIp = self.get_svc_host_ip(local_svc, self.ns)
+            clusterSvcHostIp = self.get_svc_host_ip(cluster_svc, self.ns)
+
+            localSvcExternalIP = "8.8.8.8"
+            clusterSvcExternalIP = "4.4.4.4"
+
+            # Add external IPs to the two services. Use a few known public IPs.
+            self.add_svc_external_ips(local_svc, self.ns, [localSvcExternalIP])
+            self.add_svc_external_ips(cluster_svc, self.ns, [clusterSvcExternalIP])
+
+            local_svc_externalips_route = "%s via %s" % (localSvcExternalIP, localSvcHostIp)
+            cluster_svc_externalips_route = "%s via %s" % (clusterSvcExternalIP, clusterSvcHostIp)
+
+            # Verify that external IPs are not advertised yet.
+            retry_until_success(lambda: self.assertNotIn(local_svc_externalips_route, self.get_routes()))
+            retry_until_success(lambda: self.assertNotIn(cluster_svc_externalips_route, self.get_routes()))
+
+            # Whitelist external IPs.
+            calicoctl("""apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: BGPConfiguration
+metadata:
+  name: default
+spec:
+  serviceClusterIPs:
+  - cidr: 10.96.0.0/12
+  serviceExternalIPs:
+  - cidr: %s/32
```

Comment:
Looks like these tests only test `/32` addresses here. I suspect most people will use a range.

e.g., allocate all of `172.24.16.0/24` for external IPs and then assign them out from that.